client/src/components/UpLoadImages/index.tsx

```diff
@@ -1,27 +1,44 @@
-import { Button } from '@mui/material';
+import { CloudUpload } from '@mui/icons-material';
+import { Button, CircularProgress } from '@mui/material';
 import {
   ref, getDownloadURL, uploadBytes,
 } from 'firebase/storage';
 import { useRef, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
 import storage from '../../firebase/firebaseConfig';
+import httpInstance from '../../services/axiosConfig';
 import CustomizedSnackbars from '../snackbar';
+import './style.css';
 
-function UploadFiles() {
+function UploadFiles({ carId }:{ carId:string | undefined }) {
   const [file, setFile] = useState<FileList | null>();
+  const navigate = useNavigate();
   const [loading, setLoading] = useState(false);
   const [open, setOpen] = useState(false);
   const [
     snackData,
     setSnackData,
   ] = useState<{ type: 'error' | 'success' | 'info', message: string }>({ type: 'info', message: '' });
 
+  const saveImages = async (urls: Array<string>) => {
+    const images:Array<object> = [];
+    await urls.map((url) => images.push({ image: url, carId }));
```

Comment:
why you're using `map` to push inside, 
Please check [map in js](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map)

Reply:
And why you `await` map?